arcgis-ios-sdk-samples/Maps/Generate offline map/GenerateOfflineMapViewController.swift

```diff
@@ -62,98 +62,77 @@ class GenerateOfflineMapViewController: UIViewController, AGSAuthenticationManag
         let portal = AGSPortal.arcGISOnline(withLoginRequired: true)
         
         //portal item for web map
-        self.portalItem = AGSPortalItem(portal: portal, itemID: "acc027394bc84c2fb04d1ed317aac674")
+        let portalItem = AGSPortalItem(portal: portal, itemID: "acc027394bc84c2fb04d1ed317aac674")
+        self.portalItem = portalItem
         
         //map from portal item
-        let map = AGSMap(item: self.portalItem)
+        let map = AGSMap(item: portalItem)
         
         //assign map to the map view
-        self.mapView.map = map
+        mapView.map = map
         
         //disable the bar button item until the map loads
-        self.mapView.map?.load { [weak self] (error) in
+        mapView.map?.load { [weak self] (error) in
             
-            guard error == nil else {
-                
-                //show error
-                SVProgressHUD.showError(withStatus: error!.localizedDescription)
+            if let error = error{
+                if (error as NSError).code != NSUserCancelledError{
+                    //show error
+                    SVProgressHUD.showError(withStatus: error.localizedDescription)
+                }
+                return
+            }
+            
+            guard let strongSelf = self else{
```

Comment:
This should probably be the first thing done by the completion handler. If `self` is `nil`, there's no point in showing an error.